private/lib/Service/CategoryService.php

```diff
@@ -43,24 +46,55 @@ public function getAllCategoriesForUser(int $userId): array
         return $this->categoryHelper->getAllCategoriesForUser($user);
     }
 
-    public function getCategoryForUser(int $categoryId, int $userId): CategoryModel
+    public function getAllCategoriesWithExcludeFilter(int $userId, array $excludeCategoryNames): array
     {
-        return $this->categoryHelper->getCategoryForUser($categoryId, $userId);
+        $user = $this->userHelper->getUserById($userId);
+        $categories = $this->categoryHelper->getAllCategoriesForUser($user);
+
+        $filteredCategories = [];
+        foreach ($categories as $category) {
+            if (!in_array($category->getName(), $excludeCategoryNames, true)) {
+                $filteredCategories[] = $category;
+            }
+        }
+
+        return $filteredCategories;
+    }
+
+    public function getCategoryForUser(int $categoryId, int $userId): CategoryDecorator
```

Comment:
This method returns an entry count and template count. I understand why you do this, but I am not sure if this ithe right method name for this response. Maybe rename this to `getCategoryAndStatisticalDataForUser()`

Reply:
I'm passing data to the blade file, I don't understand that we need to specify getting statistical data. It's like getting the User, all the information belongs to category, why do we need to specify something like this getCategoryWithDescription ?

Reply:
@aliselcukkaya hmm we indeed to the same for the user decorator. Looking at the code from a pull request gives me a different perspective. I think that I would have done it differently now because the name is not totally correct. But nevertheless, it makes sense to leave it for now. 

I will perform some refactoring in the future for this.